app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingFragmentTest.kt

```diff
@@ -480,21 +509,37 @@ class OnboardingFragmentTest {
   }
 
   @Test
-  // TODO(#973): Fix OnboardingFragmentTest
-  @Ignore
   fun testOnboardingFragment_clickOnSkip_changeOrientation_titleIsCorrect() {
     launch(OnboardingActivity::class.java).use {
       onView(withId(R.id.skip_text_view)).perform(click())
+      testCoroutineDispatchers.runCurrent()
       onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
       onView(
         allOf(
           withId(R.id.slide_title_text_view),
-          isDisplayed()
+          isCompletelyDisplayed()
         )
       ).check(matches(withText(R.string.onboarding_slide_3_title)))
     }
   }
 
+  private fun scrollToPage(position: Int): ViewAction {
+    return object : ViewAction {
+      override fun getDescription(): String {
+        return "Scroll ViewPager2 to position: $position"
+      }
+
+      override fun getConstraints(): Matcher<View> {
+        return isAssignableFrom(ViewPager2::class.java)
+      }
+
+      override fun perform(uiController: UiController?, view: View?) {
+        (view as ViewPager2).setCurrentItem(position, false)
+      }
+    }
+  }
```

Comment:
I couldn't find a way to remove `allOf` so tried replacing swiping by implementing a custom ViewAction that does the scrolling. But even after using this, the tests were failing, then after trying different things, I disabled smooth scrolling, and the test cases passed.

Reply:
We've had issues with smooth scrolling in the past, too (this is what prompted me to suggest using RecyclerView in the first place). I'd expect Espresso to properly stabilize against smooth scrolling. Does this mean we simply cannot use swipe for navigation & have to force the page directly?

Reply:
> Ah nice find. We've had issues with smooth scrolling in the past, too (this is what prompted me to suggest using RecyclerView in the first place). I'd expect Espresso to properly stabilize against smooth scrolling. Does this mean we simply cannot use swipe for navigation & have to force the page directly?

@BenHenning Yes, it seems like we have to force the page directly, because in this case maybe, views are not getting recycled the way they get while swiping.